### PR TITLE
fix(request-id): if the header_name is already present in the…

### PR DESCRIPTION
**Pull request by yujintang**
if the header_name is already present in the request uuid_val include_in_response should keep source

### What this PR does / why we need it:
in request-id plugin, when header_name is already present in the request, we should keep source id 

Fix https://github.com/apache/apisix/issues/6066

### Pre-submission checklist:
no

### CODE REVIEW

**Review by spacewander — COMMENTED** [review]
Could you add a test in https://github.com/apache/apisix/blob/master/t/plugin/request-id.t?
We can use https://github.com/apache/apisix/blob/e9fab8b2e5b5cb79a302faf8b5c9752766681c63/t/plugin/grpc-web.t#L164-L165 to send request header

**Comment by tzssangglass** [comment]
LGTM

File: apisix/plugins/request-id.lua [hunk 1]
```diff
@@ -214,6 +214,8 @@ function _M.rewrite(conf, ctx)
     local uuid_val = get_request_id(conf.algorithm)
     if not headers[conf.header_name] then
         core.request.set_header(ctx, conf.header_name, uuid_val)
+    else
+        uuid_val = headers[conf.header_name]
```
Comment (starsz) [inline apisix/plugins/request-id.lua:218]: I think we can avoid calling get_request_id if the id exists.

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by tokers — APPROVED** [review]

File: apisix/plugins/request-id.lua (hunk 1 above)
Comment (haoyang1994) [reply apisix/plugins/request-id.lua:219]: This change should be submitted with an switch. In my situation, i can't trust the request id I received and must rewrite it with a new uuid.

**Pull request merged**